docs: dagster_gcp.gcs.sensor.get_gcs_keys code example

mlarose opened the pull request:
## Summary & Motivation

Add documentation to get_gcs_keys.

## How I Tested These Changes

code example derived from what was used to test #20191

erinkcochran87 commented on python_modules/libraries/dagster-gcp/dagster_gcp/gcs/sensor.py:
```diff
@@ -5,7 +5,52 @@
-    """Credits to @ajohnson5 for the original implementation of this method."""
+    """Return a list of updated keys in a GCS bucket.
+
+    Credits to @ajohnson5 for the original implementation of this method.
```
This feels odd to have in the docs. Can we add it as a code comment or in the PR instead?

mlarose replied:
I can, i did credit the author in the PR by linking to his and his name will also show up as co-author of the first commit.

erinkcochran87 reviewed (commented):
Thanks for adding this! I assume these are supposed to show up in the API docs?

mlarose commented:
> Thanks for adding this! I assume these are supposed to show up in the API docs?

Yes, it should show up similar to this: https://docs.dagster.io/_apidocs/libraries/dagster-aws#s3

mlarose commented:
![Screenshot 2024-03-13 at 9 51 09 AM](https://github.com/dagster-io/dagster/assets/4839980/8c7aedb9-f26a-487b-ae96-73aa45fcc895)

Finally got sphinx to work.

erinkcochran87 reviewed (approved):
Approving to unblock ya, but I noticed something small in the rendered API docs.

In the screenshot below, the `prefix` has only one double quote (`prefix="`), but in the code itself it looks right. I'm not sure if this is a docstring issue or a rendering issue, but if ya don't mind taking another look at it I'd appreciate it!

<img width="750" alt="Screenshot 2024-03-13 at 1 01 04 PM" src="https://github.com/dagster-io/dagster/assets/16615212/b874006c-42d8-445c-95ba-2678ff026a26">

mlarose commented:
> In the screenshot below, the prefix has only one double quote (prefix="), but in the code itself it looks right. I'm not sure if this is a docstring issue or a rendering issue, but if ya don't mind taking another look at it I'd appreciate it!


weird. sphinx or autodoc converts the double quotes to single quotes. it's valid code, but misleading, since you saw it as a double quote. I changed the default string param to None and adjusted the code and unit tests to match.
